Added support for Get information for the current session's user

ParthaI opened the pull request:
Get information for the current session's user: https://api.aiven.io/doc/#tag/Users/operation/UserInfo

byashimov commented on me.go:
```diff
@@ -0,0 +1,66 @@
+	Features struct {
+		FreeTierEnabled       bool `json:"free_tier_enabled,omitempty"`
+		ReferralEnabled       bool `json:"referral_enabled,omitempty"`
+		ShowConfigDetailsStep bool `json:"show_config_details_step,omitempty"`
```
Please use `*bool` here to know if values was privided.

byashimov commented on me_test.go:
```diff
@@ -0,0 +1,155 @@
+		return false
+	}
+	return true
+}
```
Let's remove the test. It actually tests the struct and mocked value 🙏

byashimov commented on me.go:
```diff
@@ -0,0 +1,66 @@
+		CreateTime             string                 `json:"create_time,omitempty"`
+		Department             *string                `json:"department,omitempty"`
+		Features               Features               `json:"features,omitempty"`
+		Intercom               Intercom               `json:"intercom,omitempty"`
```
1. Please use a pointer for `Features`
2. It might have more fields, as it has no strict schema
3. `Intercom` is required field

byashimov commented on me.go:
```diff
@@ -0,0 +1,66 @@
+		TokenValidityBegin     *string                `json:"token_validity_begin,omitempty"`
+		User                   string                 `json:"user,omitempty"`
+		UserID                 string                 `json:"user_id,omitempty"`
+		ViewedIndicators       []string               `json:"viewed_indicators,omitempty"`
```
Some fields here are required.

byashimov reviewed (changes requested):
Hey! Thanks for your contribution!
If there is no particular reason to go with this client, I recommend to try our generated. Here is the method you are implementing https://github.com/aiven/go-client-codegen/blob/259be870943791d517994df16b18e601f0278ebe/handler/user/user.go#L107
It is experimental, but since it is generated it is more accurately than this one.

ParthaI commented:
Hi @byashimov,

Thank you for your detailed feedback. In our project, we have used the Go package `aiven-go-client`. The [suggested API](https://github.com/aiven/go-client-codegen/blob/259be870943791d517994df16b18e601f0278ebe/handler/user/user.go#L107) is in the package `go-client-codegen`. Changing the Go package in our project might break our existing implementation. 

If we could consider this PR, it would be greatly appreciated.


> Please use `*bool` here to know if values was provided.

I have  updated the `Features` type to [map[string]any](https://github.com/aiven/go-client-codegen/blob/259be870943791d517994df16b18e601f0278ebe/handler/user/user.go#L724C25-L724C39)

> Let's remove the test. It actually tests the struct and mocked value 🙏

Removed the test file.

> 1. Please use a pointer for `Features`
> 2. It might have more fields, as it has no strict schema
> 3. `Intercom` is required field

Updated the `Features` attribute type and marked the attributes as required based on the API reference in [the documentation](https://api.aiven.io/doc/#tag/Users/operation/UserInfo).

Thanks!

byashimov reviewed (changes requested):
Hey. Thanks for the updates. 
Please take a look at these:
1. there is no `viewed_indicators` field https://api.aiven.io/doc/#tag/Users/operation/UserInfo
2. should remove pointers from required field

I recommend to copy this struct https://github.com/aiven/go-client-codegen/blob/main/handler/user/user.go#L718
It was generated from the spec

ParthaI commented:
Hello @byashimov, thank you for your suggestions. I've made the changes accordingly. Could you please take another look?

byashimov commented:
Thank you @ParthaI for the contribution!
